environment/src/main/java/com/sequenceiq/environment/environment/flow/creation/handler/StorageConsumptionCollectionSchedulingHandler.java

```diff
@@ -0,0 +1,122 @@
+
+    protected StorageConsumptionCollectionSchedulingHandler(
+            EventSender eventSender,
+            EnvironmentService environmentService,
+            EventBus eventBus,
+            EntitlementService entitlementService,
+            ConsumptionService consumptionService,
+            @Value("${environment.consumption.enabled:false}") boolean consumptionEnabled) {
+        super(eventSender);
+        this.environmentService = environmentService;
+        this.eventBus = eventBus;
+        this.entitlementService = entitlementService;
+        this.consumptionService = consumptionService;
+        this.consumptionEnabled = consumptionEnabled;
+    }
+
+    @Override
+    public void accept(Event<EnvironmentDto> environmentDtoEvent) {
+        LOGGER.debug("Storage consumption collection scheduling flow step started.");
+        EnvironmentDto environmentDto = environmentDtoEvent.getData();
+        try {
+            environmentService.findEnvironmentById(environmentDto.getId())
+                    .ifPresentOrElse(environment -> {
+                                scheduleStorageConsumptionCollectionIfNeeded(environmentDto);
+                                goToNextState(environmentDtoEvent, environmentDto);
+                            }, () -> goToFailedState(environmentDtoEvent, environmentDto,
+                                    new IllegalStateException(String.format("Environment was not found with id '%s'.", environmentDto.getId())))
+                    );
+        } catch (Exception e) {
+            LOGGER.error("Storage consumption collection scheduling failed", e);
+            goToFailedState(environmentDtoEvent, environmentDto, e);
+        }
+        LOGGER.debug("Storage consumption collection scheduling flow step finished.");
+    }
+
+    private void scheduleStorageConsumptionCollectionIfNeeded(EnvironmentDto environmentDto) {
+        String accountId = environmentDto.getAccountId();
+        if (consumptionEnabled && entitlementService.isCdpSaasEnabled(accountId)) {
+            scheduleStorageConsumptionCollection(environmentDto);
+        } else {
+            LOGGER.info("Skipping storage consumption collection scheduling because " +
+                    (consumptionEnabled ? String.format("CDP_SAAS entitlement is missing for account '%s'", accountId) : "it is disabled for the deployment"));
+        }
+    }
+
+    private void scheduleStorageConsumptionCollection(EnvironmentDto environmentDto) {
+        StorageConsumptionRequest request = new StorageConsumptionRequest();
+        String resourceCrn = environmentDto.getResourceCrn();
+        request.setEnvironmentCrn(resourceCrn);
+        request.setMonitoredResourceCrn(resourceCrn);
+        request.setMonitoredResourceName(environmentDto.getName());
+        request.setMonitoredResourceType(ResourceType.ENVIRONMENT);
+        request.setStorageLocation(environmentDto.getTelemetry().getLogging().getStorageLocation());
```

Comment:
What if telemetry is null? (Of course this is a test edge case only but for example I use to create my environments without telemetry logging)

A second question: what about backup storage location?

Reply:
Good points!
* The log location is mandatory for env creation. It is a required parameter for CDP CLI, and also a mandatory input field in the UI wizard. I wonder if this can ever be `null`. Perhaps only for `MOCK` or `YARN` providers? I have no idea TBH.
* The backup storage location is an interesting one. I am not sure we touched upon that if it also needs to be measured.

@nadamdb Could you please chime in regarding the second bullet point? :smile:

Reply:
One note: The backup location defaults to the log location unless overridden. So if we were to decide on reporting consumption for the backup location as well, the schedule & unschedule calls would have to take this into consideration (and skip the backup location if it matches the log one).

Reply:
Just to make a note about it here: as discussed, backup location scheduling will be added later as that location is not static and can be changed after env creation so we need a bit different approach there. 
The scheduling would work similarly to log location scheduling but the unscheduling is a bit trickier and we have to prevent accidentally unscheduling the log location in cases where the backup bucket was originally the same as the log one but is later changed to a different one (as the crn of the monitored resource and the storage location together uniquely identify a consumption collection, hence if both are the same we only schedule it once).

Reply:
I will add support for the backup location scheduling/unscheduling now, but full support for the backup location edit scenario will be tackled later in [CB-17410](https://jira.cloudera.com/browse/CB-17410).